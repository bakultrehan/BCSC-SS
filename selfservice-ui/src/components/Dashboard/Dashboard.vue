/** * Dashboard of app */

<template>
  <v-card class="mx-auto" style="max-width: 80%;">
    <v-toolbar flat class="bc-subtitle" dark>
      <v-toolbar-title>{{ $t('dashboard.pagetitle') }}</v-toolbar-title>
      <div class="flex-grow-1"></div>
    </v-toolbar>
    <v-divider></v-divider>
    <v-container>
      <v-row class="ma-2">
        <v-col cols="12">
          <v-card-subtitle
            class="text-left bc-padding-left-0 page-info pa-2"
            v-html="$t('dashboard.createProjectInfo')"
          ></v-card-subtitle>
        </v-col>
        <Button
          class="white--text ml-3 yellow-btn"
          depressed
          :yellowBtn="true"
          @click="$router.push(`/project/info`)"
          name="btn-create-project"
          data-test-id="btn-create-project"
        >{{ $t('dashboard.btnCreateProject') }}</Button>
        <v-col cols="12" md="12" v-if="projectInfoList.length > 0">
          <h2 class="bc-page-title-h1 px-2 mb-2">{{ $t('dashboard.myprojectTitle') }}</h2>

          <v-simple-table class="text-left">
            <template v-slot:default>
              <thead class="table-head">
                <tr>
                  <th
                    :scope="$t('dashboard.tblTitleProjectName')"
                  >{{ $t('dashboard.tblTitleProjectName') }}</th>
                  <th
                    :scope="$t('dashboard.tblTitleReferenceNo')"
                  >{{ $t('dashboard.tblTitleReferenceNo') }}</th>
                  <th :scope="$t('dashboard.tblTitlCreated')">{{ $t('dashboard.tblTitlCreated') }}</th>
                  <th
                    :scope="$t('dashboard.tblTitlrole')"
                    v-if="isClient"
                  >{{ $t('dashboard.tblTitlrole') }}</th>
                  <th
                    :scope="$t('dashboard.tblTitleProjectStatus')"
                  >{{ $t('dashboard.tblTitleProjectStatus') }}</th>
                </tr>
              </thead>
              <tbody>
                <tr
                  v-for="project in projectInfoList"
                  :key="project.id"
                  @click="redirectToProject(project)"
                  style="cursor: pointer"
                >
                  <td>{{ project.name }}</td>
                  <td>{{ project.id }}</td>
                  <td>{{ project.created }}</td>
                  <td v-if="isClient">{{ project.role }}</td>

                  <td>
                    {{
                    $t(`dashboard.role${projectStatusList[project.statusId]}`)
                    }}
                  </td>
                </tr>
              </tbody>
            </template>
          </v-simple-table>
        </v-col>
        <v-col cols="12" md="12" v-else>
          <v-card flat class="text-left">
            <v-card-text>{{ $t('dashboard.noData') }}</v-card-text>
          </v-card>
        </v-col>
      </v-row>
    </v-container>
  </v-card>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';
import { Getter, namespace, Action } from 'vuex-class';
import { projectStatus } from '@/constants/enums';

import Button from '@/Atomic/Button/Button.vue';

const ProjectInfoModule = namespace('ProjectInfoModule');
const KeyCloakModule = namespace('KeyCloakModule');
const SharedModule = namespace('SharedModule');

@Component({
  components: {
    Button
  }
})
export default class Dashboard extends Vue {
  @KeyCloakModule.Getter('isClient')
  public isClient!: any;

  @ProjectInfoModule.Getter('getProjectInfoList')
  public projectInfoList!: any;
  @ProjectInfoModule.Action('loadProjectInfo')
  public loadProjectInfo!: any;
  @ProjectInfoModule.Action('errorStatus')
  public errorStatus!: any;
  @SharedModule.Action('redirectFromSummaryPage')
  public redirectFromSummaryPage!: any;

  @SharedModule.Getter('isRedirectFromSummaryPage')
  public isRedirectFromSummaryPage!: boolean;

  private projectStatusList: any = projectStatus;

  private mounted() {
    this.loadProjectInfo();
  }

  private redirectToProject(project: any) {
    if (project.statusId === projectStatus.development) {
      this.redirectFromSummaryPage(false);
    }
    this.$router.push(`/project-container/${project.id}`);
  }
}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';
.table-head {
  background-color: $BCgovBlue5 !important;
  color: $BCgovWhite !important;
  & tr {
    & th {
      color: $BCgovWhite !important;
      font-size: 14px;
    }
  }
}
</style>
