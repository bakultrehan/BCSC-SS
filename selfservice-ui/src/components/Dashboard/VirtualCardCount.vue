/** * Dashboard of app */

<template>
  <v-card class="mx-auto">
    <v-toolbar flat class="bc-subtitle" dark>
      <v-toolbar-title>{{
        $t('dashboard.virtualCardCountTitle')
      }}</v-toolbar-title>
      <div class="flex-grow-1"></div>
    </v-toolbar>
    <v-divider></v-divider>
    <v-container>
      <v-row class="ma-2">
        <v-col cols="12" md="8">
          <RemaingAccounts />
        </v-col>
        <v-col cols="12" md="4">
          <div class="remaining-acc-btn">
            <Button
              class="white--text ml-3"
              depressed
              @click="$router.push(`/add-test-account`)"
              name="btn-create-project"
              data-test-id="btn-create-project"
              >{{ $t('dashboard.btnManageTestAccount') }}</Button
            >
          </div>
        </v-col>
      </v-row>
    </v-container>
  </v-card>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import { namespace } from 'vuex-class';

import Button from '@/Atomic/Button/Button.vue';
import RemaingAccounts from '@/components/TestAccount/RemaingAccounts.vue';

const TestAccountModule = namespace('TestAccountModule');

@Component({
  components: {
    Button,
    RemaingAccounts,
  },
})
export default class VirtualCardCount extends Vue {}
</script>

<style lang="scss" scoped>
@import './../../assets/styles/theme.scss';

.remaining-acc-btn {
  width: 250px;
  display: inline;
  float: right;
}
</style>
