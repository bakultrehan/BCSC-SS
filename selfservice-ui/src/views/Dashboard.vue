<template>
  <div class="home">
    <Dashboard />
  </div>
</template>

<script lang="ts">
// @ is an alias to /src
import { Component, Vue, Prop } from 'vue-property-decorator';
import Dashboard from '@/components/Dashboard/Dashboard.vue';

@Component({
  components: {
    Dashboard
  }
})
export default class DashboardHome extends Vue {}
</script>
