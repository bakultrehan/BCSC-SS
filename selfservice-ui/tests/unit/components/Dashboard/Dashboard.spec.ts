import { shallowMount } from '@vue/test-utils';
import Dashboard from '@/components/Dashboard/Dashboard.vue';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
let vuetify: any;
vuetify = new Vuetify();

describe('Dashboard.vue', () => {
  const store = new Vuex.Store({
    modules: {
      ProjectInfoModule: {
        namespaced: true,
        state: {},
        getters: {
          getProjectInfoList: jest.fn(() => {
            return [];
          })
        },
        actions: {
          loadProjectInfo: jest.fn()
        }
      }
    }
  });

  const mountFunction = (options: any) => {
    return shallowMount(Dashboard, {
      store,
      vuetify,
      mocks: { $t: jest.fn(() => {}) }, // tslint:disable-line
      ...options
    });
  };

  it('renders props when passed with gettors', () => {
    const dashboard = mountFunction({});
    expect(dashboard.element).toMatchSnapshot();
  });
});
