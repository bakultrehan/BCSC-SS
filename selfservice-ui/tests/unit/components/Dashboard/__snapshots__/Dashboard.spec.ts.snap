// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard.vue renders props when passed with gettors 1`] = `
<v-card-stub
  class="mx-auto"
  loaderheight="4"
  style="max-width: 80%;"
  tag="div"
>
  <v-toolbar-stub
    class="bc-subtitle"
    dark="true"
    extensionheight="48"
    flat="true"
    src=""
    tag="header"
    tile="true"
  >
    <v-toolbar-title-stub>
      
    </v-toolbar-title-stub>
     
    <div
      class="flex-grow-1"
    />
     
    <div
      class="flex-grow-1"
    />
     
    <v-btn-stub
      activeclass=""
      class="ma-2"
      color="#fba30e"
      dark="true"
      data-test-id="btn-create-project"
      fab="true"
      tag="button"
      type="button"
    >
      <v-icon-stub
        dark=""
        large=""
      >
        mdi-plus
      </v-icon-stub>
    </v-btn-stub>
  </v-toolbar-stub>
   
  <v-divider-stub />
   
  <v-container-stub
    tag="div"
  >
    <v-row-stub
      class="ma-5"
      tag="div"
    >
      <v-col-stub
        cols="12"
        md="12"
        tag="div"
      >
        <v-card-stub
          class="text-left"
          flat="true"
          loaderheight="4"
          tag="div"
        >
          <v-card-text-stub>
            
          </v-card-text-stub>
        </v-card-stub>
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</v-card-stub>
`;
